src/dda/feature_flags/manager.py

```diff
@@ -9,6 +9,8 @@
 from functools import cached_property
 from typing import TYPE_CHECKING, Any, Optional
```

Comment:
nit: this can come in a separate PR but can we please remove all `Optional` and `Union`? Instead, we should use `... | None`.